File: backends/ruby-hanami/app/infrastructure/model/invitation.rb
# frozen_string_literal: true
#
# 招待モデルを定義するモジュール。
#
# @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

module Infrastructure
  module Model
    # 招待の永続化モデルです。
    # @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
    Invitation = Struct.new(
      :id, :token,
      :created_at, :created_by, :updated_at, :updated_by,
      :deleted_at, :deleted_by, :version,
      keyword_init: true
    )
  end
end

File: backends/go-echo/ent/schema/invitation.go
package schema

import (
	"entgo.io/ent"
	"entgo.io/ent/dialect/entsql"
	"entgo.io/ent/schema"
	"entgo.io/ent/schema/field"
	"time"
)

// Invitation holds the schema definition for the Invitation entity.
type Invitation struct {
	ent.Schema
}

func (Invitation) Annotations() []schema.Annotation {
	return []schema.Annotation{
		entsql.Annotation{Table: "invitations"},
	}
}

func (Invitation) Fields() []ent.Field {
	return []ent.Field{
		field.Uint("id").Positive().Immutable(),
		field.String("token"),
		field.Time("created_at").Default(time.Now),
		field.Uint("created_by").Optional().Nillable(),
		field.Time("updated_at").Default(time.Now),
		field.Uint("updated_by").Optional().Nillable(),
		field.Time("deleted_at").Optional().Nillable(),
		field.Uint("deleted_by").Optional().Nillable(),
		field.Int("version").Default(0),
	}
}

func (Invitation) Edges() []ent.Edge {
	return nil
}
